src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java

```diff
@@ -44,55 +44,194 @@
+
+        public Builder withRecipientType(FeedbackParticipantType recipientType) {
+            if (recipientType != null) {
+                feedbackQuestionAttributes.recipientType = recipientType;
+            }
+            return this;
+        }
+
+        public Builder withNumOfEntitiesToGiveFeedbackTo(int numOfEntitiesToGiveFeedbackTo) {
+            feedbackQuestionAttributes.numberOfEntitiesToGiveFeedbackTo = numOfEntitiesToGiveFeedbackTo;
+            return this;
+        }
+
+        public Builder withShowResponseTo(List<FeedbackParticipantType> showResponseTo) {
+            feedbackQuestionAttributes.showResponsesTo =
+                    showResponseTo == null ? new ArrayList<>()
+                            : new ArrayList<>(showResponseTo);
+            return this;
+        }
+
+        public Builder withShowGiverNameTo(List<FeedbackParticipantType> showGiverNameTo) {
+            feedbackQuestionAttributes.showGiverNameTo =
+                    showGiverNameTo == null ? new ArrayList<>()
+                            : new ArrayList<>(showGiverNameTo);
+            return this;
+        }
+
+        public Builder withShowRecipientNameTo(List<FeedbackParticipantType> showRecipientNameTo) {
+            feedbackQuestionAttributes.showRecipientNameTo =
+                    showRecipientNameTo == null ? new ArrayList<>()
+                            : new ArrayList<>(showRecipientNameTo);
+            return this;
+        }
+
+        public Builder withCreatedAt(Instant createdAt) {
+            if (createdAt != null) {
+                feedbackQuestionAttributes.createdAt = createdAt;
+            }
+            return this;
+        }
+
+        public Builder withUpdatedAt(Instant updatedAt) {
+            if (updatedAt != null) {
+                feedbackQuestionAttributes.updatedAt = updatedAt;
+            }
+            return this;
+        }
+
+        public Builder withFeedbackQuestionId(String feedbackQuestionId) {
+            if (feedbackQuestionId != null) {
+                feedbackQuestionAttributes.feedbackQuestionId = feedbackQuestionId;
+            }
+            return this;
+        }
+
+        public FeedbackQuestionAttributes build() {
+            feedbackQuestionAttributes.questionDescription =
+                    SanitizationHelper.sanitizeForRichText(feedbackQuestionAttributes.questionDescription);
+            feedbackQuestionAttributes.removeIrrelevantVisibilityOptions();
+            return feedbackQuestionAttributes;
+        }
+
+        public FeedbackQuestionAttributes buildWithoutRemovingIrrelevantVisibilityOptions() {
+            feedbackQuestionAttributes.questionDescription =
+                    SanitizationHelper.sanitizeForRichText(feedbackQuestionAttributes.questionDescription);
+            return feedbackQuestionAttributes;
+        }
     }
 
     public FeedbackQuestionAttributes getCopy() {
-        return new FeedbackQuestionAttributes(this);
+        return builder()
```

Comment:
Add `.withFeedbackQuestionId(feedbackQuestionId)`

Reply:
I wasn't sure if getting a copy of this object means the id will stay the same as well, since the deep copy would mean it's 2 objects instead of 2 pointers pointing to the same reference. 

On another note, this method doesn't seem to be used anywhere except for the test.

Reply:
If that's the case, then maybe add a javadoc that says this?

Reply:
> On another note, this method doesn't seem to be used anywhere except for the test.

Maybe I have missed something as I cannot find the usage of this method in the test.

A side note is if the production code doesn't use the method. We should not keep it just for testing.